addons/purchase/views/purchase_views.xml

```diff
@@ -814,21 +822,54 @@
         <field name="name">purchase.history.tree</field>
         <field name="model">purchase.order.line</field>
         <field name="arch" type="xml">
-            <tree create="false" default_order="order_id asc">
+            <tree create="false" default_order="order_id asc" action="action_open_order" type="object">
+                <field name="currency_id" column_invisible="True"/>
                 <field name="order_id" widget="many2one"/>
                 <field name="date_approve"/>
-                <field name="partner_id"/>
-                <field name="product_uom_qty"/>
-                <field name="price_unit"/>
+                <field name="partner_id" string="Vendor"/>
+                <field name="company_id" optional="show" groups="base.group_multi_company"/>
+                <field name="product_uom_qty" string="Quantity"/>
+                <field name="price_unit" widget="monetary"/>
+                <field name="price_subtotal" widget="monetary" string="Total Untaxed"/>
+                <field name="state" optional="hide"/>
             </tree>
         </field>
     </record>
 
+    <record id="purchase_history_pivot_view" model="ir.ui.view">
+         <field name="name">purchase.history.pivot</field>
+         <field name="model">purchase.order.line</field>
+         <field name="arch" type="xml">
+             <pivot>
+                <field name="price_total" string="Total purchased amount" type="measure"/>
+                <field name="product_uom_qty" string="Total Qty purchased" type="measure"/>
+             </pivot>
+         </field>
+    </record>
+
+     <record id="purchase_history_graph_view" model="ir.ui.view">
+         <field name="name">purchase.history.graph</field>
+         <field name="model">purchase.order.line</field>
+         <field name="arch" type="xml">
+             <graph string="purchase Analysis" type="line" sample="1">
```

Comment:
Why there is sample="1" here and not in other views? (I don't think we need sample data here who told you to add one?)

Reply:
Done the changes :+1: